src/prommis/uky/costing/ree_plant_capcost.py

```diff
@@ -284,6 +284,12 @@ def build_process_costs(
         """
 
         # define costing library
+        if hasattr(self, "library") and self.library == "REE":  # costing already exists
+            raise Exception(
```

Comment:
1. You should avoid using bare Exceptions (and pylint will complain about these). `RunTimeError` or maybe `AttributeError` would be better.
2. f-strings are generally faster than string formatting, so I would encourage you to use them instead of this.